tensorly/decomposition/_cp.py

```diff
@@ -73,8 +74,19 @@ def initialize_cp(tensor, rank, init='svd', svd='numpy_svd', random_state=None,
 
     elif isinstance(init, (tuple, list, CPTensor)):
         # TODO: Test this
-        try:
+        try:         
+            if normalize_factors is True:
+                warnings.warn('It is not recommended to initialize an initial tensor with normalizing. Consider normalizing the initial tensor before using this function')
+            
             kt = CPTensor(init)
+            weights, factors = kt
+            
+            if tl.all(weights == 1):
+                kt = CPTensor((None, factors))
+            else:
+                for i in range(len(factors)):
+                    factors[-1] = factors[-1]*weights[i]
```

Comment:
Actually, there is no specific reason for that. Scaling only one factor doesn't effect the reconstruction. If you prefer, I would change to scale all the factors with related weights.

Reply:
It could even be a separate function in that case, e.g. to make all factors equal weights? Do you think it would make sense to check the numerical stability when i) absorbing all weights in one factors and ii) spreading them in all factors?

Reply:
As we agreed last week, we updated factor initialization by spreading weights in all factors. In order to avoid unstable weight values, we update all the factors with geometric mean of weight values. I hope current version is ok for merging